radicale/storage.py

```diff
@@ -364,17 +364,41 @@ def create_collection(cls, href, collection=None, tag=None):
             self.set_meta("tag", "VCALENDAR")
             if collection:
                 collection, = collection
+                items = []
                 for content in ("vevent", "vtodo", "vjournal"):
-                    if content in collection.contents:
-                        for item in getattr(collection, "%s_list" % content):
-                            new_collection = vobject.iCalendar()
-                            new_collection.add(item)
-                            self.upload(uuid4().hex, new_collection)
+                    items.extend(getattr(collection, "%s_list" % content, []))
+                processed_uids = []
+                for i, item in enumerate(items):
+                    uid = getattr(item, "uid", None)
+                    if uid in processed_uids:
+                        continue
+                    new_collection = vobject.iCalendar()
+                    new_collection.add(item)
+                    if uid:
+                        processed_uids.append(uid)
+                        # search for items with same UID
+                        for oitem in items[i+1:]:
```

Comment:
This is `O(n^2)`. It could be `O(n)` by having a dict that maps from UIDs to `iCalendar` objects.